contracts/TradeFactory/TradeFactoryPositionsHandler.sol

```diff
@@ -172,26 +154,34 @@ abstract contract TradeFactoryPositionsHandler is ITradeFactoryPositionsHandler,
     emit TradesCanceled(msg.sender, _canceledTradesIds);
   }
 
-  function changePendingTradesSwapper(string memory _swapper)
+  function setStrategySwapper(
+    address _strategy,
+    address _swapper,
+    bool _migrateSwaps
+  ) external override returns (uint256[] memory _changedSwapperIds) {
+    _setStrategySwapper(_strategy, _swapper);
+    if (_migrateSwaps) {
+      return _changeStrategyPendingTradesSwapper(_strategy, _swapper);
+    }
+  }
```

Comment:
**VULNERABILITY:** Anyone could execute this with `_migrateSwaps=false` since `_setStrageySwappers` does not validate any roles.